### DISCUSSION TITLE
Setting session token in pg_net?

### DISCUSSION THREAD

**Discussion by daniel-krastev** [supabase/supabase#28909] [Questions] [answered]
Hi there,

I have an edge function which I want to call from a postgres trigger. In that function I want to get the user that calls it. For that reason I want to get the session token of the calling user. Is it possible to do that from the trigger? 

Thanks!

**Comment by TheOtherBrian1** [comment]
When PostgREST (the DB API) makes a request, it creates identifying Postgres variables for the duration of the query. You can use the `auth.uid()` function to get the user id, which is actually just the following code under-the-hood:

```sql
 select 
  coalesce(
    nullif(current_setting('request.jwt.claim.sub', true), ''),
    (nullif(current_setting('request.jwt.claims', true), '')::jsonb ->> 'sub')
  )::uuid
```

I would like to emphasize that this will only work when triggered by the anon or authenticated role.

To see how to incorporate it into pg_net, you can checkout the [docs](https://supabase.com/docs/guides/database/extensions/pg_net?queryGroups=database-method&database-method=sql#invoke-a-supabase-edge-function)

**Comment by daniel-krastev** [comment]
My question was more about setting the token in the `"Authorization": "Bearer <USER_TOKEN>"` header, so in my edge function I can do something like this?
```
	const authHeader = req.headers.get('Authorization')
	const token = authHeader?.replace('Bearer ', '')
	const { data, error } = await supabaseClient.auth.getUser(token)
```

**Reply by TheOtherBrian1** [reply, marked as answer]
You can view the session variables with the following database function ([must be called through the DB AP](https://supabase.com/dashboard/project/_/api?page=rpc-intro)I):

```sql
CREATE OR REPLACE FUNCTION get_current_settings(
) RETURNS JSONB AS $$
DECLARE
    settings JSONB;
BEGIN
    -- Retrieve the settings as JSONB
    settings := jsonb_build_object(
        'search_path', current_setting('search_path', true),
        'role', current_setting('role', true),
        'request.jwt.claims', current_setting('request.jwt.claims', true),
        'request.method', current_setting('request.method', true),
        'request.path', current_setting('request.path', true),
        'request.headers', current_setting('request.headers', true),
        'request.cookies', current_setting('request.cookies', true)
    );

    RETURN settings;
END;
$$ LANGUAGE plpgsql;
```

I believe the `request.headers` variable contains the full JWT under the `authorization` key. However, it may make more sense to [call the edge function directly from the browser](https://supabase.com/docs/guides/functions/cors)

**Comment by daniel-krastev** [comment]
Thank you very much! I can see the headers there!

But why do you think it would be better to do that from the browser? I mean, that is trigger calling an edge function after insert on my users table, where in the edge function I need to create an associated stripe customer. Having that outside of the browser would mean it would also work if I want to make a flutter app let say.
